### DISCUSSION TITLE
How to obtain the logs from function of the app when a task is running using celery?

### DISCUSSION THREAD

**Discussion by pulse-mind** [cookiecutter/cookiecutter-django#5559] [Q&A] [answered]
I have a function on a model that write logs at INFO level. 
When I call this function using Django Admin (for example), then I can see the logs in the file. 
It is on a production server. I am using supervisor, in Django I log into console, in the supervisor configuration I set a stdout file.

When I call this function in a celery task, the log of this function is not written in any log. I can see the log from celery
```Task.... received```. Celery is setup in a supervisor with a stdout too. 

The project was setup using cookiecutter.

I tried many things that I found on Google but nothing work.

**Comment by browniebroke** [comment]
Yea, I think we may be missing some config here. These posts seem to suggest that we need to do some setup using Celery signals:

- https://dnmtechs.com/django-celery-logging-best-practices-for-python-3-programming/
- https://stackoverflow.com/questions/48289809/celery-logger-configuration

There is a similar setup done in one of the projects I use, to configure the formatter to structured loging format

**Comment by pulse-mind** [comment]
Thank you for your answer. 

I already set in my code : https://stackoverflow.com/questions/48289809/celery-logger-configuration, the answer about dictConfig(settings.LOGGING)

Thank you for the first link  https://dnmtechs.com/django-celery-logging-best-practices-for-python-3-programming/
I Added 
```CELERYD_HIJACK_ROOT_LOGGER = False```
And I set 
```'disable_existing_loggers': False,``` because it was True for me.

And it works :)

**Comment by pulse-mind** [comment]
May be it could be nice to improve cookiecutter with these inputs ?

**Reply by browniebroke** [reply]
Yes, definitely. PR welcome

**Reply by pulse-mind** [reply, marked as answer]
done https://github.com/cookiecutter/cookiecutter-django/pull/5563
